server/sonar-web/src/main/js/apps/settings/licenses/LicenseChangeForm.js

```diff
@@ -18,32 +18,89 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 import React from 'react';
-import LicenseValueView from './LicenseValueView';
-import { translate } from '../../../helpers/l10n';
+import Modal from 'react-modal';
+import { translate, translateWithParameters } from '../../../helpers/l10n';
 
 export default class LicenseChangeForm extends React.PureComponent {
   static propTypes = {
     license: React.PropTypes.object.isRequired,
     onChange: React.PropTypes.func.isRequired
   };
 
+  state = {
+    loading: false,
+    modalOpen: false
+  };
+
   onClick(e) {
     e.preventDefault();
     e.target.blur();
+    this.setState({ modalOpen: true });
+  }
 
-    const { license, onChange } = this.props;
+  closeModal = () => this.setState({ modalOpen: false });
 
-    new LicenseValueView({
-      productName: license.name || license.key,
-      value: license.value,
-      onChange
-    }).render();
-  }
+  handleSubmit = event => {
+    event.preventDefault();
+    if (this.textarea) {
+      const { value } = this.textarea;
+      this.setState({ loading: true });
+      this.props
+        .onChange(value)
+        .then(
+          () => this.setState({ loading: false, modalOpen: false }),
+          () => this.setState({ loading: false })
+        );
+    }
+  };
+
+  handleCancelClick = event => {
+    event.preventDefault();
+    this.closeModal();
+  };
 
   render() {
+    const { license } = this.props;
+    const productName = license.name || license.key;
+
     return (
       <button className="js-change" onClick={e => this.onClick(e)}>
         {translate('update_verb')}
+
+        {this.state.modalOpen &&
+          <Modal
+            isOpen={true}
+            contentLabel="modal"
+            className="modal"
+            overlayClassName="modal-overlay"
+            onRequestClose={this.closeModal}>
+            <form onSubmit={this.handleSubmit}>
+              <div className="modal-head">
+                <h2>{translateWithParameters('licenses.update_license_for_x', productName)}</h2>
+              </div>
+              <div className="modal-body">
+                <label htmlFor="license-input">{translate('licenses.license_input_label')}</label>
+                <textarea
+                  autoFocus={true}
+                  className="width-100 spacer-top"
+                  ref={node => (this.textarea = node)}
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract the assignment of "this.textarea" from this expression. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AAssignmentWithinCondition)